project-service/api.md

```diff
@@ -0,0 +1,594 @@
+- Data Params
+  + Required
+    * `name=[string]`
+    * `status=[string]`
+    * `startDate=[date]`
+    * `endDate=[date]`
+- Success Response
+  + Code: `201 CREATED`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'extra features',
+  status: 'Not Started',
+  startDate: [date],
+  endDate: [date],
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Edit Sprint
+
+Edit details of a sprint
+
+- URL
+  + `/sprints`
+- Method
+  + `PUT`
+- URL Params
+  + Required
+    * `sprintId=[number]`
+- Data Params
+  + Optional
+    * `name=[string]`
+    * `status=[string]`
+    * `startDate=[date]`
+    * `endDate=[date]`
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'extra features',
+  status: 'Started',
+  startDate: [date],
+  endDate: [date],
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Sprint
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Sprint doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Delete Sprint
+
+Delete a sprint (tasks that belong to this sprint are not deleted)
+
+- URL
+  + `/sprints`
+- Method
+  + `DELETE`
+- URL Params
+  + Required
+    * `sprintId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `204 NO CONTENT`
+  + Content: None
+- Error Response
+  + Invalid Sprint
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Sprint doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+## Tasks
+
+### Fetch Tasks for a Project
+
+Get all tasks for a project
+
+- URL
+  + `/tasks`
+- Method
+  + `GET`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 1,
+    name: 'do a task',
+    description: 'Ready',
+    userId: 2,
+    sprintId: 4
+  }, {
+    id: 2,
+    name: 'do another task',
+    description: 'In Progress',
+    userId: 3,
+    sprintId: 4
+  }, {
+    id: 4,
+    name: 'do yet another task',
+    description: 'Done',
+    userId: null,
+    sprintId: null
+  }
+]
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Create Task for a Project
+
+Create a new task for a project
+
+- URL
+  + `/tasks`
+- Method
+  + `POST`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + Required
+    * `name=[string]`
+    * `description=[string]`
+    * `status=[string]`
```

Comment:
Since we had discussed keeping a hardcoded list of statuses on the frontend server, the client side should be in charge of setting the default status. When a user creates a task, just send a POST request with status always set to "Not Started". That makes more sense to me.